### PR TITLE
Serverless Platform Updates

### PR DESCRIPTION
**Pull request by eahefnawy**
- [x] Refresh tokens if they expire
- [x] Add deployment data
- [x] Add initial subscriptions data
- [x] Add resources data

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/createStack.js [hunk 1]
```diff
@@ -40,7 +41,17 @@ module.exports = {
       'CloudFormation',
       'createStack',
       params
-    ).then((cfData) => this.monitorStack('create', cfData));
+    ).then((cfData) => this.monitorStack('create', cfData)).catch((e) => {
+      if (this.serverless.service.deployment && this.serverless.service.deployment.deploymentId) {
+        const deploymentData = this.serverless.service.deployment;
+        deploymentData.status = 'Failed';
+        return platform.createDeployment(deploymentData)
```
Comment (pmuens) [inline lib/plugins/aws/deploy/lib/createStack.js]: Shouldn't this be `updateDeployment` and the `createDeployment` is called before the CloudFormation template is even handed over to AWS?

File: lib/plugins/aws/deploy/lib/createStack.js [hunk 2]
```diff
@@ -87,6 +98,8 @@ module.exports = {
             .then(this.create);
         }
 
+        // update deployment failure
```
Comment (pmuens) [inline lib/plugins/aws/deploy/lib/createStack.js]: Is this a TODO?

File: lib/plugins/aws/lib/updateStack.js [hunk 3]
```diff
@@ -45,7 +46,17 @@ module.exports = {
     return this.provider.request('CloudFormation',
       'createStack',
       params)
-      .then((cfData) => this.monitorStack('create', cfData));
+      .then((cfData) => this.monitorStack('create', cfData)).catch((e) => {
+        if (this.serverless.service.deployment && this.serverless.service.deployment.deploymentId) {
+          const deploymentData = this.serverless.service.deployment;
+          deploymentData.status = 'Failed';
+          return platform.createDeployment(deploymentData)
```
Comment (pmuens) [inline lib/plugins/aws/lib/updateStack.js]: Same here. Shouldn't we create a new deployment beforehand with the status `in-progress` (this is the default one if no status is provided) and then update it here with the status of `failed`?

File: lib/plugins/aws/lib/updateStack.js [hunk 4]
```diff
@@ -95,7 +106,15 @@ module.exports = {
       .then((cfData) => this.monitorStack('update', cfData))
       .catch((e) => {
         if (e.message === NO_UPDATE_MESSAGE) {
-          return;
+          return BbPromise.resolve();
+        }
+        if (this.serverless.service.deployment && this.serverless.service.deployment.deploymentId) {
+          const deploymentData = this.serverless.service.deployment;
+          deploymentData.status = 'Failed';
+          return platform.createDeployment(deploymentData)
```
Comment (pmuens) [inline lib/plugins/aws/lib/updateStack.js]: And here as well.

File: package.json [hunk 5]
```diff
@@ -91,8 +91,7 @@
     "sinon-chai": "^2.9.0"
   },
   "dependencies": {
-    "@serverless/fdk": "^0.5.1",
-    "@serverless/platform-sdk": "^0.1.5",
+    "@serverless/platform-sdk": "file:../platform-sdk",
```
Comment (pmuens) [inline package.json]: Looks like this needs to be updated before it's merged.

**Comment by ac360** [comment]
`package.json` `paltform-sdk` dependency is still pointing to a local file.

**Pull request merged**